src/nbl/video/COpenGLConnection.h

```diff
@@ -14,17 +13,19 @@ class COpenGLConnection final : public IAPIConnection
 public:
     COpenGLConnection()
     {
-        // "offscreen" for windowless app
-        //actually fuck SDL2
-        // global state for whole process
-        SDL_VideoInit("windows");
+        
     }
 
 protected:
     ~COpenGLConnection()
     {
 
     }
+
+private:
+    // Note: EGL is not initialized here, each thread willing to use EGL will need to initialize it separately
+    // (thats at least how our spoof EGL behaves, not sure if it's how EGL spec wants it)
+    egl::CEGLCaller m_egl;
```

Comment:
so why keep it in the API connection and not some per-thread/per-context data structure?

Reply:
so that pointers are loaded only once, API connection passes CEGLCaller pointer down the hierarchy (physical device->logical device->queue)

Reply:
there's actually a little known quirk that extension (not core) function pointers loaded using one context (via `GetProcAddr`) are not safe to use in another

Reply:
although thats mostly applicable to the OpenGL and OpenGL ES APIs, not sure about EGL itself

Reply:
> there's actually a little known quirk that extension (not core) function pointers loaded using one context (via GetProcAddr) are not safe to use in another

you're talking about GL func pointers, those are EGL
Regarding GL pointers... currently each object of COpenGLLogicalDevice has its own pointers however (+ table of available extensions/features/limits etc), however theres no problem to move it to COpenGLQueue or some per-thread struct